packages/flutter_goldens_client/lib/skia_client.dart

```diff
@@ -12,38 +12,72 @@ import 'package:path/path.dart' as path;
-import 'package:flutter_goldens_client/client.dart';
-
 // If you are here trying to figure out how to use golden files in the Flutter
 // repo itself, consider reading this wiki page:
 // https://github.com/flutter/flutter/wiki/Writing-a-golden-file-test-for-package%3Aflutter
 
-// TODO(Piinks): This file will replace ./client.dart when transition to Skia
-// Gold testing is complete
-
+const String _kFlutterRootKey = 'FLUTTER_ROOT';
 const String _kGoldctlKey = 'GOLDCTL';
 const String _kServiceAccountKey = 'GOLD_SERVICE_ACCOUNT';
 
-/// An extension of the [GoldensClient] class that interfaces with Skia Gold
-/// for golden file testing.
-class SkiaGoldClient extends GoldensClient {
-  SkiaGoldClient({
-    FileSystem fs = const LocalFileSystem(),
-    ProcessManager process = const LocalProcessManager(),
-    Platform platform = const LocalPlatform(),
-  }) : super(
-    fs: fs,
-    process: process,
-    platform: platform,
-  );
+/// A client for uploading image tests and making baseline requests to the
+/// Flutter Gold Dashboard.
+class SkiaGoldClient {
+  SkiaGoldClient(
+    this.workDirectory, {
+    this.fs = const LocalFileSystem(),
+    this.process = const LocalProcessManager(),
+    this.platform = const LocalPlatform(),
+    io.HttpClient httpClient,
+  }) : assert(workDirectory != null),
+       assert(fs != null),
+       assert(process != null),
+       assert(platform != null),
+       httpClient = httpClient ?? io.HttpClient();
+
+  /// The file system to use for storing the local clone of the repository.
+  ///
+  /// This is useful in tests, where a local file system (the default) can
+  /// be replaced by a memory file system.
+  final FileSystem fs;
+
+  /// A wrapper for the [dart:io.Platform] API.
+  ///
+  /// This is useful in tests, where the system platform (the default) can
+  /// be replaced by a mock platform instance.
+  final Platform platform;
+
+  /// A controller for launching sub-processes.
+  ///
+  /// This is useful in tests, where the real process manager (the default)
+  /// can be replaced by a mock process manager that doesn't really create
+  /// sub-processes.
+  final ProcessManager process;
+
+  /// A client for making Http requests to the Flutter Gold dashboard.
+  final io.HttpClient httpClient;
 
   /// The local [Directory] within the [comparisonRoot] for the current test
   /// context. In this directory, the client will create image and json files
   /// for the goldctl tool to use.
   ///
   /// This is informed by the [FlutterGoldenFileComparator] [basedir]. It cannot
   /// be null.
-  Directory _workDirectory;
+  final Directory workDirectory;
+
+  /// A map of known golden file tests and their associated positive image
+  /// hashes.
+  ///
+  /// This is set and used by the [FlutterLocalFileComparator] and
+  /// [FlutterPreSubmitFileComparator] to test against golden masters maintained
+  /// in the Flutter Gold dashboard.
+  Map<String, dynamic> get expectations => _expectations;
```

Comment:
dynamic in this API is kinda sad. Can we strongly type this? Maybe the API here should be that there's a method getExpactationFor(foo) and it just returns a List of Strings (hashes)?

Reply:
Fixed! :)